tools/shared.py

```diff
@@ -1625,13 +1633,13 @@ def nop(arg):
     # emcc.py etc. The unsuffixed versions have the python_selector logic that can
     # pick the right version as needed (which is not crucial right now as we support
     # both 2 and 3, but eventually we may be 3-only).
-    env['CC'] = quote(unsuffixed(EMCC)) if not WINDOWS else 'python %s' % quote(EMCC)
-    env['CXX'] = quote(unsuffixed(EMXX)) if not WINDOWS else 'python %s' % quote(EMXX)
-    env['AR'] = quote(unsuffixed(EMAR)) if not WINDOWS else 'python %s' % quote(EMAR)
-    env['LD'] = quote(unsuffixed(EMCC)) if not WINDOWS else 'python %s' % quote(EMCC)
+    env['CC'] = quote(unsuffixed(EMCC)) if not WINDOWS or MSYS else 'python %s' % quote(EMCC)
+    env['CXX'] = quote(unsuffixed(EMXX)) if not WINDOWS or MSYS else 'python %s' % quote(EMXX)
+    env['AR'] = quote(unsuffixed(EMAR)) if not WINDOWS or MSYS else 'python %s' % quote(EMAR)
+    env['LD'] = quote(unsuffixed(EMCC)) if not WINDOWS or MSYS else 'python %s' % quote(EMCC)
     env['NM'] = quote(LLVM_NM)
-    env['LDSHARED'] = quote(unsuffixed(EMCC)) if not WINDOWS else 'python %s' % quote(EMCC)
-    env['RANLIB'] = quote(unsuffixed(EMRANLIB)) if not WINDOWS else 'python %s' % quote(EMRANLIB)
+    env['LDSHARED'] = quote(unsuffixed(EMCC)) if not WINDOWS or MSYS else 'python %s' % quote(EMCC)
+    env['RANLIB'] = quote(unsuffixed(EMRANLIB)) if not WINDOWS or MSYS else 'python %s' % quote(EMRANLIB)
```

Comment:
Are you saying that under msys its possible to launch "emcc.py" directly with python's subprocess?

Are are you relying on the above subprocess hack?

Reply:
This is unrelated to the subprocess/ENOEXEC hack/workaround.

The comment above these lines does explain the issue here:

> some configure scripts expect e.g. CC to be a literal executable (but "python emcc.py" is not a file that exists). 

So, with this change i am enforcing plain unsuffixed/literal commands are used under MSYS.